src/cheri_cap_common.sail

```diff
@@ -466,7 +466,14 @@ function getCapCursor(cap) : Capability -> CapAddrInt = unsigned(cap.address)
 function int_to_cap (offset) : CapAddrBits -> Capability =
     {null_cap with address = offset}
 
-function invalidateCap(cap) : Capability -> Capability = {cap with tag=false}
+function clearTagIf(cap, cond) : (Capability, bool) -> Capability =
+    {cap with tag = cap.tag & not(cond)}
+
+function clearTagIfSealed(cap) : Capability -> Capability =
+    clearTagIf(cap, isCapSealed(cap))
+
+function clearTag(cap) : Capability -> Capability =
+    clearTagIf(cap, true)
```

Comment:
Not sure if this extra indirection makes the code more readable, I'd just go for
```suggestion
    {cap with tag = false}
```